### DISCUSSION TITLE
Limit chats retrieved

### DISCUSSION THREAD

**Discussion by guillezcurra** [tomtau/superbasic-im#4] [Q&A] [answered]
Hello, do you know if there is any way to limit the chats in client.getChats() ? I have way too many chats and it would be nice to see only "Recent chats", maybe the last 20. Because when it tries to load all chats it takes a lot of time.

Thanks!

**Comment by tomtau** [comment]
Unfortunately it seems no... at least as far as I know, maybe there's a way. whatsapp-web.js only has a method to load all chats. Generally, the first load is the slowest, because it needs to decrypt all of them, and it's faster later.

Loading contacts is faster, but there's no data about which one has the more recent chat.

**Reply by guillezcurra** [reply]
Okay, thank you. I have been using the web-app for a while now and it does get faster but not really enough, it takes for me almost 1 minute to load the chats page.

**Reply by tomtau** [reply, marked as answer]
1 minute seems long, so not sure how many is "way too many" chats and where it could possibly be from rendering the page (instead of loading the chats from whatsapp-web.js). You can try to add a line here: https://github.com/tomtau/superbasic-im/blob/main/src/routes.ts#L86 to cut the number of chats to e.g. 20 (` chats = chats.slice(0, 20);`), rebuild the project and see if it helps.

**Reply by guillezcurra** [reply]
I did your suggested change and it worked! I counted and before it was 20 seconds to load (sorry I was too dramatic saying 1 min) and now it takes 3 seconds, I have created two options in my main page "Recent chats" and "All Chats" and it works great. I can share the code if you want.

Thank you!

**Reply by tomtau** [reply]
@guillezcurra awesome that it worked!

>  I can share the code if you want.

Sure, feel free to open a PR. The steps for contributing code are described here: https://github.com/tomtau/superbasic-im/blob/main/CONTRIBUTING.md#working-on-issues

I opened this issue: https://github.com/tomtau/superbasic-im/issues/5

**Comment by guillezcurra** [comment]
<img src="https://github.com/user-attachments/assets/78b9324a-e98d-45f3-a142-af9cce1a9ad5" width="300" height="400">
